heft-plugins/heft-sass-plugin/src/SassTypingsGenerator.ts

```diff
@@ -133,45 +181,43 @@ export class SassTypingsGenerator extends StringValuesTypingsGenerator {
     buildFolder: string,
     importIncludePaths: string[] | undefined
   ): Promise<string> {
-    const result: Result = await LegacyAdapters.convertCallbackToPromise(render, {
-      data: fileContents,
-      file: filePath,
-      importer: (url: string) => ({ file: this._patchSassUrl(url) }),
-      includePaths: importIncludePaths
-        ? importIncludePaths
-        : [path.join(buildFolder, 'node_modules'), path.join(buildFolder, 'src')],
-      indentedSyntax: path.extname(filePath).toLowerCase() === '.sass'
-    });
+    const result: Result = await new Promise(
+      (resolve: (result: Result) => void, reject: (err: Error) => void) => {
+        render(
+          {
+            data: fileContents,
+            file: filePath,
+            importer: (url: string) => ({ file: this._patchSassUrl(url) }),
+            includePaths: importIncludePaths
+              ? importIncludePaths
+              : [path.join(buildFolder, 'node_modules'), path.join(buildFolder, 'src')],
+            indentedSyntax: path.extname(filePath).toLowerCase() === '.sass'
+          },
+          (err: SassError, result: Result) => {
+            if (err) {
+              // Extract location information and format into the error message until we have a concept
+              // of location-aware diagnostics in Heft.
+              return reject(new Error(`${err.file}(${err.column},${err.line}): ${err.message}`));
+            }
+            resolve(result);
+          }
+        );
+      }
+    );
 
     // Register any @import files as dependencies.
-    const target: string = result.stats.entry;
     for (const dependency of result.stats.includedFiles) {
-      this.registerDependency(target, dependency);
+      this.registerDependency(filePath, dependency);
```

Comment:
This appears like a subtle and correct change, however, I'm not sure it is best. 
E.g. If we have a set of Sass files with @include statements such that:
A.sass -> B.sass -> C.sass
and C.sass is changed, do we want to start at the top (A.sass, which I believe is result.stats.entry) and work our way down, or do we want to start at B.sass (the next parent filePath) and compile and then trigger A.sass (which will work down the dependency to B.sass anyway).

I'm not sure this will generate extra work, but I think it is worth investigating.

Reply:
`result.stats.entry` and `filePath` are expected to refer to the same file, except that since the values will be compared via exact string comparisons it is better to use the value passed into the function as the key in the table, in order to ensure that we find the file in the dependency table.